andela/ah-robotics-frontend
# #162949045  Article tags should be present in all relevant views

#### What does this PR do?

Enables a user to add tags to an article and view them when the article loads

#### Description of Task to be completed?
- create a tag input
- enable tags on article creation
- display tags when article loads

#### How should this be manually tested?
1. Navigate to a folder where you would prefer to test the repository.

2. Create a directory
 ```$```  ```mkdir myproject```

3. Navigate to the directory "myproject"
 ```$```   ```cd myproject```
4. Clone the author's haven repository
  ```$```   ```git clone https://github.com/andela/ah-robotics-frontend.git```
5. Change the directory into the "ah-robotics-fronted"
  ```$```   ```cd ah-robotics-frontend```

6.  Checkout to the branch ft-tags-162949045
 ```$```   ```git checkout ft-tags-162949045```

7. Install dependancies from the package.json
 ```$```   ```npm install``` to install all the modules listed in the `package.json` file.

8. Start the application in development mode
  ```$```   ```npm start``` to run the app in the development mode 

The application opens http://localhost:3000 in the browser.
Change the URL on the browser to;
 ```
https://ah-robotics-frontend-sta-pr-19.herokuapp.com
```

The homepage will appear as shown below.
![image](https://user-images.githubusercontent.com/14871749/53410237-42ea1d80-39d4-11e9-9abb-d6b9b620b88d.png)


9. Click the ```Login``` link on the Navbar and enter the following details to log in:
 **(for demo purpose use)**

**Username:** ```o.ogeto@yahoo.com``` 
**Password:** ```Ogeto1234$```

10. Once you are logged in, click ```New Article```. You will be presented with a page to create a new article. Enter your article content. On the tags space, insert tags and press enter after each tag. The screen should appear as shown below.
![image](https://user-images.githubusercontent.com/14871749/53717410-2719c800-3e69-11e9-8a0d-7bdcff0aeb98.png)


11. Press ```Post``` button on the top right side of the page. The article will be created and you will be redirected to the given article. Check the tags below the article image.
![image](https://user-images.githubusercontent.com/14871749/53717371-02255500-3e69-11e9-8317-28259a4837c0.png)

12. To run tests, open the terminal and run the following command:
```
npm test
``` 
13. To run tests and coverage, open the terminal and run the following command:
```
npm test -- --coverage
```
#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#162949045](https://www.pivotaltracker.com/story/show/162949045)

#### Screenshots (if appropriate)
##### Login a user
<img width="1280" alt="screen shot 2019-02-22 at 09 46 27" src="https://user-images.githubusercontent.com/14871749/53224753-d515bd00-3686-11e9-9aec-94eddaea401e.png">

##### Articles page
![image](https://user-images.githubusercontent.com/14871749/53717472-516b8580-3e69-11e9-9149-98a7a0fae271.png)


#### Article with tags:
![image](https://user-images.githubusercontent.com/14871749/53717371-02255500-3e69-11e9-8317-28259a4837c0.png)


#### Questions:
